### PR TITLE
Remove original device preprocessing (new API)

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
The device preprocessing from lightning is currently not compatible with Catalyst.

**Description of the Change:**
This PR adds the discard of the original preprocessing.

**Benefits:**
Easier support for Lightning.

**Possible Drawbacks:**
Device specific preprocessing is ignored.

### CODE REVIEW

**Comment by rmoyard** [comment]
[sc-60560]

File: frontend/catalyst/qjit_device.py [hunk 1]
```diff
@@ -294,7 +295,9 @@ def preprocess(
         execution_config: qml.devices.ExecutionConfig = qml.devices.DefaultExecutionConfig,
     ):
         """Device preprocessing function."""
-        program, config = self.original_device.preprocess(execution_config)
+        # TODO: readd the user preprocessing program once transforms are compatible with TOML files
```
Comment (erick-xanadu) [inline frontend/catalyst/qjit_device.py:298]:
```suggestion
        # TODO: read the user preprocessing program once transforms are compatible with TOML files
```

File: frontend/test/lit/test_device_api.py [hunk 2]
```diff
@@ -86,7 +86,8 @@ def circuit():
 
 def test_preprocess():
     """Test a circuit (with preprocessing transforms) compilation to MLIR when
-    using the new device API."""
+    using the new device API.
+    TODO: we need to readd the two check-not once we accept the device preprocessing."""
```
Comment (erick-xanadu) [inline frontend/test/lit/test_device_api.py:90]:
```suggestion
    TODO: we need to read the two check-not once we accept the device preprocessing."""
```

File: frontend/test/pytest/test_device_api.py [hunk 3]
```diff
@@ -79,15 +79,16 @@ def test_qjit_device():
     # Check the preprocess of the new device
     transform_program, _ = device_qjit.preprocess()
     assert transform_program
-    assert len(transform_program) == 3
+    assert len(transform_program) == 2
 
-    t = transform_program[0].transform.__name__
-    assert t == "split_non_commuting"
+    # TODO: readd when we do not discard user preprocessing
```
Comment (erick-xanadu) [inline frontend/test/pytest/test_device_api.py:84]:
```suggestion
    # TODO: read when we do not discard user preprocessing
```

**Review by erick-xanadu — APPROVED** [review]
:partying_face:

File: frontend/catalyst/qjit_device.py (hunk 1 above)
Comment (rmoyard) [reply frontend/catalyst/qjit_device.py:298]: That's what I meant https://en.wiktionary.org/wiki/readd

File: frontend/test/pytest/test_device_api.py (hunk 3 above)
Comment (rmoyard) [reply frontend/test/pytest/test_device_api.py:84]: see above

File: frontend/test/lit/test_device_api.py (hunk 2 above)
Comment (rmoyard) [reply frontend/test/lit/test_device_api.py:90]: see above

File: frontend/catalyst/qjit_device.py (hunk 1 above)
Comment (rmoyard) [inline frontend/catalyst/qjit_device.py:298]:
```suggestion
        # TODO: readd the device preprocessing program once transforms are compatible with TOML files
```

File: frontend/test/pytest/test_device_api.py (hunk 3 above)
Comment (rmoyard) [inline frontend/test/pytest/test_device_api.py:84]:
```suggestion
    # TODO: readd when we do not discard device preprocessing
```

**Pull request merged**